tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/WebRequestCommon.cs

```diff
@@ -55,11 +55,14 @@ public static CallTargetState GetResponse_OnMethodBegin<TTarget>(TTarget instanc
                         if (setSamplingPriority)
                         {
                             scope.Span.SetTraceSamplingPriority(spanContext.SamplingPriority.Value);
-                            scope.Span.Context.TraceContext.LockSamplingPriority();
+                            scope.Span.LockSamplingPriority();
                         }
 
                         // add distributed tracing headers to the HTTP request
-                        SpanContextPropagator.Instance.Inject(scope.Span.Context, request.Headers.Wrap());
+                        if (scope.Span.Context is SpanContext scopeSpanContext)
```

Comment:
While this check should always return true for our automatic instrumentation, we still need it for type safety. In the unusual case that this check doesn't pass, is it okay if we lock the sampling priority still? Or should this check be appended to the `scope != null` check?